### ISSUE TITLE
Previewing large PDFs freezes Waterfox

### ISSUE THREAD

**Issue by flolilo** [BrowserWorks/waterfox#733]
**Describe the bug**
If you open a (large enough) PDF in Waterfox's preview, Waterfox will permanently freeze.

**To Reproduce**
1. Have Waterfox set to preview PDF-files (default)
2. Open any larger PDF in-browser (preferably in a new tab). [Here's an example of a file](http://gdlp01.c-wss.com/gds/0/0300024970/01/eos5d-mk4-im-en.pdf) It does not matter if the PDF was already visited before or not.

Now, the load-circle in the PDF's tab-header is spinning ad infinitum, while all other tabs freeze (i.e. you cannot write into text boxes any more). Closing the PDF-tab does not un-freeze Waterfox. Task Manager does not show any network activity at all.

Other browsers like Firefox and Vivaldi do not show this behavior.

**Edit:** Now something interesting has happened: While I can reproduce this behavior when opening a link to a file at duckduckgo.com or stackexchange.com, I cannot reproduce it when opening my example file from this very issue. Perhaps it would be best, then, to look up any camera's manual (e.g. `Canon EOS 5D Mark IV manual`) in DuckDuckGo to reproduce this issue.

**Edit 2: What I already tried:**
 - Safe mode _(no change)_ 
 - Creating a new profile _(no change)_
 - (De)activating plug-ins _(no change)_
 - Deactivating my Antivirus _(no change)_

**Expected behavior**
PDF should load, Waterfox should not freeze (obviously ;-) )

**Screenshots**
Can deliver a screen capture, if you need that.

**Desktop (please complete the following information):**
 - OS: Windows 10 Pro v1803 x64, latest updates. Multiple systems with different hardware configurations are subject to this bug.
 - Windows Defender as A/V
 - Waterfox: 5.6.2.2 & 5.6.2.3 (64 bit)

**Comment by grahamperrin** [comment]
Initial load of http://gdlp01.c-wss.com/gds/0/0300024970/01/eos5d-mk4-im-en.pdf in 56.2.2 on FreeBSD-CURRENT: 

- no problem. 

From the properties dialogue: 

- File size: 36.1 MB (37,840,304 bytes)
- Creator: FrameMaker 6.0
- PDF Producer: Acrobat Distiller 6.0.1 (Windows)
- PDF Version: 1.5
- Page Count: 676 

Is the issue reproducible with Waterfox in safe mode?

**Comment by flolilo** [comment]
Yes, the problem persists in Safe Mode.

As said, it is interesting that I cannot reproduce it reliably when I click on my own link here at GitHub, while on every other site I use (and click on PDFs) this problem is there 100% of the time. It is not even specific to this PDF - any PDF that has more than, say, a few MB will lead to a permanent freeze.

**Comment by grahamperrin** [comment]
Thanks. Reproducible with a new profile? 

- about:profiles

**Comment by flolilo** [comment]
Just tested this now - same behavior as with the default profile.
As said, this problem occurs on all computers that I maintain - all have Win10 v1803 x64 installed, but here, the similarities end.

Thanks for the great support!

**Comment by Squall-Leonhart** [comment]
antivirus url filtering

**Comment by grahamperrin** [comment]
about:plugins

What's there?

**Comment by flolilo** [comment]
- **OpenH264-Videocodec** 1.7.1
 - **Widevine Content Decryption Module** 1.4.8.1008
 - **AdobeExManDetect** 1.1.0.0 _(deactivated)_
 - **WacomTabletPlugin** 2.1.0.7 _(deactivated)_
 - **VLC Web Plugin** 3.0.3.0

As to Antivirus: I am using Windows's Defender. Tried deactivating it just now: **No difference in behavior.**

**Comment by grahamperrin** [comment]
> … **Edit**: … cannot reproduce it when opening my example file …

This reminds me of a recent _Firefox_ for Windows (10) case that ended as inexplicably as it began: 

- [Firefox PDF Preview not working for already downloaded files : firefox](https://www.reddit.com/r/firefox/comments/9n2ys0/firefox_pdf_preview_not_working_for_already/)

Try this: 

1. about:preferences#applications
2. for PDF, prefer anything **other** than _Preview in Waterfox_
3. exit
4. start Waterfox
5. reopen a previously affected PDF (how, exactly, do you choose to reopen?)
6. about:preferences#applications
7. for PDF, revert to _Preview in Waterfox_
8. exit
9. start Waterfox
10. reopen the previously affected PDF.

Is it previewed? 

If not: 

- for test purposes, create a new profile. 

----

Defocusing from Waterfox (which I began using only a few months ago) and PDF: over the years I have lost count of the number of times that Firefox stubbornly refused to respect application handling preferences. Maybe more often bugged on Windows than on other platforms. 

I lost track of relevant bugs (sorry) but I did get a sense that Mozilla fixed a few things. Maybe this Waterfox case has remnants of one of those bugs. 

When you first used Waterfox, can you recall, did you accept the invitation to import from Firefox?

**Comment by flolilo** [comment]
@grahamperrin There is no difference in trying it with already downloaded files or new ones. Also, that reopen-sequence (I tried that before, but without knowing about this issue) does not help.

> When you first used Waterfox, can you recall, did you accept the invitation to import from Firefox?

No, as I have not even had installed Firefox at the time of installing Waterfox - I just tried to reproduce this issue with it later on. Also, I did not import anything from other browsers.

Note: I updated the initial post, so that people do not have to read through all of it again.
